facebook/OpenBIC
# fby3.5: rf: Support write/read CXL controller

Summary:
- Support write and read CXL controller flash via SPI.

Test plan:
- Build code: Pass
- Write CXL flash controller: Pass
- Read CXL flash  controller: Pass

Log:
1. Check CXL flash can write and read.
- Initial SPI
[BMC console]
root@bmc-oob:~# bic-util slot3 0xe0 0x02 0x9c 0x9c 0x00 0x05 0xe0 0x09 0x9c 0x9c 0x00 0x03 0x00 0x00 0x00 0x00 0x01 0x00 0x00
9C 9C 00 05 39 09 00 9C 9C 00

[BIC console]
uart:~$ [00:02:52.594,000] <inf> spi_aspeed: Flash data is monotonous, skip calibration.
[00:02:52.594,000] <inf> spi_aspeed: Flash data is monotonous, skip calibration.

- Read CXL flash
uart:~$ flash read spi1_cs0 fffe0 20
000FFFE0: ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff |........ ........|
000FFFF0: ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff |........ ........|

- Write CXL flash
uart:~$ flash write spi1_cs0 fffe0 20 11111111 22222222
Write OK.
Verified.
uart:~$ flash read spi1_cs0 fffe0 20
000FFFE0: 20 00 00 00 11 11 11 11  22 22 22 22 ff ff ff ff | ....... """"....|
000FFFF0: 22 22 22 22 ff ff ff ff  ff ff ff ff ff ff ff ff |"""".... ........|
